### PR TITLE
[TD-11946]<feature>: parse block data and handle null value

### PR DESCRIPTION
**Pull request by huolibo**

### CODE REVIEW

File: src/connector/jdbc/src/main/java/com/taosdata/jdbc/utils/NullType.java [hunk 1]
```diff
@@ -11,81 +11,97 @@ public String toString() {
...
     }
-    
+
     public static boolean isTinyIntNull(byte val) {
         return val == Byte.MIN_VALUE;
     }
-    
+
+    public static boolean isUnsignedTinyIntNull(byte val) {
+        return val == (byte) 0xFF;
```
Comment (sangshuduo) [inline src/connector/jdbc/src/main/java/com/taosdata/jdbc/utils/NullType.java:20]: we should use taos_is_null() from now.

File: src/connector/jdbc/src/main/java/com/taosdata/jdbc/utils/NullType.java [hunk 2]
```diff
@@ -11,81 +11,97 @@ public String toString() {
...
+    }
+
     public static boolean isSmallIntNull(short val) {
         return val == Short.MIN_VALUE;
     }
 
+    public static boolean isUnsignedSmallIntNull(short val) {
+        return val == (short) 0xFFFF;
```
Comment (sangshuduo) [inline src/connector/jdbc/src/main/java/com/taosdata/jdbc/utils/NullType.java:28]: same concern

**Review by sangshuduo — CHANGES_REQUESTED** [review]
see my comments

**Review by dingbo8128 — APPROVED** [review]
LGTM

**Review by sangshuduo — APPROVED** [review]
LGTM

**Pull request merged**